### DISCUSSION TITLE
Dictionary

### DISCUSSION THREAD

**Discussion by outlogic** [plateaukao/einkbro#161] [Ideas]
eBook reading apps allow to launch external dictionaries like Fora Dictionary, AARD, GoldenDict,... or online translators like Google Translate. It would be nice to have this option in the browser (for example, launching a dictionary after double tapping on the word).

**Comment by plateaukao** [comment]
For more reader-app prone features, you could save web content as epub file, and then you can read the article in your preferred reading app. 
To support external dictionaries by double tapping on the word, I may need to hack all web site content, and inject actions into the web site. As web sites are developed in different ways, this would be error-prone, unlike ebook format, which is more standardized and easier to manipulate. If you found any open-source browser that supports this feature, please let me know. I would love to take a look at it.

**Reply by ekuke** [reply]
Actually I could translate a word in Onyx Boox note air 2. When I hold word apears meniu in which I choose Dictionary. And opens Onyx Boox Dictionary app with selected word translation. My interest is it possible adapt in some way that i could mark a sentence and choose not Dictionary app, but Google translate app?

**Reply by plateaukao** [reply]
Yes. If you installed Google Translate App, it should appear in the menu as the first action.

**Comment by ekuke** [comment]
It's strange thant only in EinkBrowser option "Translate" doesn't appear
(In other apps apears for example firefox browser). Maybe it's some how
related with settings and apps I have installed.

**Reply by plateaukao** [reply]
Are you using the latest version of EinkBro?

**Comment by ekuke** [comment]
Hi, thanks for you replay. I'm using lates version of Einkbro (v.18.0).
Today I tried a few experiments. Actually "Translate" option appears in
firefox allways, but in Einkbro it appears only then webpage language is
not English, for example it apears then I open Lithuanian webpages (I'm
from Lithuania). And that's ironical, because mostly I need translation
from English to Lithuania. Maybe do you have any ideas what to do to make
"Translate" option to appear indepedently from webpage language?

**Reply by plateaukao** [reply]
Wow...I found the same issue on my device too!
Do you set your device system language setting to English?
If so, could you try to temporarily change it to Lithuanian, and check if "Translate" option appears on English content?
I think this might be a feature(issue) implemented by Google Translate, or ONYX BOOX devices.

**Comment by ekuke** [comment]
Ok. I'm attaching two screenshots. In the second one screenshot button with
three vertical dots also doesn't give Translate option.

**Comment by ekuke** [comment]
Yes, my device system language is set to english.
I have tried your suggested experiment and temporarily changed system
language to russian (Lithuanian language is not supported by Onyx Boox).
Translate button doesn't appear in russian language, but apears in others
(Lithuanian, English). So this could be a workaround. But I still intrested
why Translate button allways apears in Firefox independently from system
language. Btw I have tried EinkBro on my regular android phone (English is
set as system language) and Translate button also appears independently
from system language. Seems like webpage text rendering methods used by
EinkBrowser are implemented differently by Onyx. And those methods differs
from Firefox. But it's pure speculation.
